C-Plus-Plus/cp/rain_water_trapping.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+    To find the total amount of water collected, we
+    firstly take an input of an array of size n of 
+    heights of buildings. After that we will try 
+    to find out minimum heights upto which water
+    could be stored at that particulat index of building.
+    This process has to be done from both left and the 
+    right sides. 
+    The complexity of the whole process will be O(t*n)
+    where, t stands for total number of test cases and n 
+    stands for total number of buildings in each test-case
+
+*/
+#include<bits/stdc++.h>
+using namespace std;
+
+// Function to return amount of water trapped in each test case
+void solve()
+{
+    //taking input of number of buildings
+    long long int n;
+    cin>>n;
+    // Making an array v containing heights of buildings
+    long long int v[n];
+    for(long long int i=0;i<n;i++)cin>>v[i];
+
+    // Making a left and a right array. 
+    long long int left_arr[n],right_arr[n];
+    left_arr[0]=v[0];
+    right_arr[n-1]=v[n-1];
+    // Computing subsequent max elements from left sidewards.
+    for(long long int i=1;i<n;i++)
+    {
+        left_arr[i]=max(v[i],left_arr[i-1]);
+    }
+    // Computing subsequent max elements from right sidewards.
+    for(long long int i=n-2;i>=0;i--)
+    {
+        right_arr[i]=max(right_arr[i],v[i+1]);
+    }
+    long long int sum=0;
+
+    // To find the amount of water getting collected inbetween the Buildings:
+    // We find minimum between the right and the left array at same indices, 
+    // Then Subtract the amount of space taken by the building at that particular index.
+
+    for(long long int i=0;i<n;i++)
+    {
+        //Take sum of collected water at each index.
+        sum+=min(left_arr[i],right_arr[i])-v[i];
+    }
+    cout<<sum<<"\n";
+}
+
+// Driver Code
+int main()
+{
+    ios_base::sync_with_stdio(false);
+    cin.tie(NULL);
+    // Taking input of number of test cases
+    long long int t;
+    cin>>t;
+    while(t--)
+    {    
+        // Calling void function
+        solve();
+    }
```

Comment:
Remove the no of test case part as we are not using online judge so no need of that used main as driver code . Also write output in the same way your code is producing. use proper display message how you are reading and displaying outputs what is `n` doesn't make sense. Use cout for for displaying message.

Reply:
Sir, I have done the changes, Please review.